### DISCUSSION TITLE
audio output measured in dB

### DISCUSSION THREAD

**Discussion by rr1164** [MTG/essentia#1266] [Q&A]
I wanted to ask if there is a project that can measure the real-time audio output level in dB. there is a similar example on the essentia website that measures the audio input in dBFS but I'd like to measure the audio output not input. 
this is the example project:https://mtg.github.io/essentia.js/examples/demos/rms-rt/

**Comment by Hikari-Tsai** [comment]
Essentia measures the sample stream you give it; it does not distinguish “input” from “output.” To meter your application's output, place the Essentia AudioWorklet in the Web Audio graph immediately before `audioContext.destination`:

```text
sources -> effects/master gain -> meter worklet -> destination
```

Send the same samples passing through the worklet to Essentia.js and compute RMS/dBFS exactly as in the microphone demo. Make sure the worklet passes the samples through unchanged so it remains part of the audible graph.

For stereo, either meter each channel separately or use the intended channel combination for your display. A basic peak/RMS dBFS conversion is `20 * log10(value)`, with a small floor before `log10` to avoid `-Infinity`.

If by “audio output” you mean all sound produced by the operating system or by other applications, a browser page cannot access that signal directly. You would need an explicitly captured display/system-audio stream where the browser supports it, a loopback audio device, or a native application. But metering the output of your own Web Audio graph requires only moving the worklet to the end of that graph.
